### ISSUE TITLE
Incorrect decoding on safari 10.1.

### ISSUE THREAD

**Issue by mrcnkoba** [jerrybendy/url-search-params-polyfill#20]
Hey guys, I think I may have found a bug in a Safari's 10.1 native URLSearchParams implementation. Because of the fact the `URLSearchParams` and `Proxy` is supported in Safari 10.1 the polyfill still uses a faulty native implementation. (https://github.com/jerrybendy/url-search-params-polyfill/blob/master/index.js#L125).

### Environment
OS: MacOS Sierra
Browser: Safari 10.1

### Expected Behavior

When typing in a dev console of virtually every browser:
```new URLSearchParams("s=%2B").get("s") // "+"```

### Actual Behavior

In Safari 10.1 it yields a space.
```new URLSearchParams("s=%2B").get("s") // " "```

The solution is to to fully replace native URLSearchParams with the polyfill, because of `Proxy` the polyfill still uses a faulty native implementation. I've prepared a pull request for it. 
https://github.com/jerrybendy/url-search-params-polyfill/pull/21

**Comment by jerrybendy** [comment]
You're right. I have just tested on safari 10.1.1, and it really has something wrong on safari 10.1.1.

I tested that:

```js
var a = new URLSearchParams('a=%2B&b=+&c=t')

a.get('a')  // Should be '+', but ' '
a.get('b')  // Should be ' ', correct
a.get('c')  // Should be 't', correct
```

😯

**Comment by mrcnkoba** [comment]
Hey @jerrybendy, yes indeed! 

In the current code also the `Proxy` chooses the native implementation that has a bug in it. My take on that was to use the full Polyfill in this case, as presented in https://github.com/jerrybendy/url-search-params-polyfill/pull/21

**Comment by jerrybendy** [comment]
Hi, I've merged #21 and release a new version `2.0.3`. Thanks for your contribution 😁
